components/ai_chat/resources/distiller_scripts/scripts/x.com/utils.ts

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { LEO_DISTILLATION_LEVEL } from '../distillation'
+import { getDistillationLevel } from './distiller'
+
+export function isString(value: any) {
+  return typeof value === 'string'
+}
+
+export function isSupportedPage(document: Document) {
+  const supportedPaths = [
+    '/',
+    '/home',
+    '/brave',
+    '/bravesampson',
+    '/brendaneich',
```

Comment:
we're going to make this generic before merging?

Reply:
Indeed. Those were added to test the distillation of profile pages. I've replaced these with a small check for a known profile page header element for now.